### PR TITLE
Cleanup repo non-code

### PR DESCRIPTION
**Pull request by yangshun**
- Added Code of Conduct to better adhere to [Community Standards](https://github.com/nusmodifications/nusmods/community)
- Removed outdated CONTRIBUTING
- Hide confusing and outdated v2 and vagrant instructions
- Add more ways to contact us

### CODE REVIEW

File: LICENSE-MIT [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Copyright (c) 2016 NUSModifications
+Copyright (c) 2017 NUSModifications
```
Comment (li-kai) [inline LICENSE-MIT]: 2018? Might as well right.

**Comment by ZhangYiJiang** [comment]
Could hold this for a bit? I want to update v3 README with the folder structure updates and some info on using CSS, thanks.

**Comment by yangshun** [comment]
Absolutely. Lemme know when it's ok. Feel free to add on to this PR.

File: LICENSE-MIT (hunk 1 above)
Comment (taneliang) [reply LICENSE-MIT]: "2016-2018" maybe?

Comment (yangshun) [reply LICENSE-MIT]: I'll update all the copyright years together in one shot on 31st Dec (:

Gonna update to `2012-Present`

**Pull request merged**